File: 2239. Find Closest Number to Zero/Solution2239.py
class Solution2239(object):
    def findClosestNumber(self, nums):
        closest = nums[0]
        for i in range(len(nums)):
            if (abs(nums[i]) < abs(closest)):
                closest = nums[i]
        if (closest < 0 and abs(closest) in nums):
            return abs(closest)
        else:
            return closest
        

File: 2239. Find Closest Number to Zero/Solution2239.java
class Solution2239 {

    public static int findClosestNumber(int[] nums) {
        boolean containSameButPositive = false;
        int closest = nums[0];
        for (int i = 0; i < nums.length; i++) {

            if (Math.abs(nums[i]) < Math.abs(closest))
                closest = nums[i];
        }

        for (int i : nums) {

            if (i == -closest) {
                containSameButPositive = true;
                break;
            }
        }

        if (closest < 0 && containSameButPositive) {
            return ((int) Math.abs(closest));
        } else {
            return (int) closest;
        }

    }
}

public static void main(String[] args) {
    int[] nums = { 2, 3, -1, -4, 2 };
    System.out.println(Solution2239.findClosestNumber(nums));
}
